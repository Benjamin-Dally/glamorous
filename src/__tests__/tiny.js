/* eslint func-style:0, react/prop-types:0 */
import React from 'react'
import {render} from 'enzyme'

import glamorousTiny from '../tiny'

test('should pass glam object prop', () => {
  const dynamicStyles = jest.fn(({glam: {big}}) => ({
    fontSize: big ? 20 : 10,
  }))
  const Comp = glamorousTiny('div')(
    {
      marginLeft: '24px',
    },
    dynamicStyles,
  )

  const glam = {big: true}
  const theme = {color: 'blue'}
  expect(
    render(<Comp id="hey-there" glam={glam} theme={theme} />),
  ).toMatchSnapshot()
  expect(dynamicStyles).toHaveBeenCalledTimes(1)
  const props = {
    glam,
    id: 'hey-there',
    theme,
  }
  const context = expect.any(Object) // the context
  expect(dynamicStyles).toHaveBeenCalledWith(props, context)
})
