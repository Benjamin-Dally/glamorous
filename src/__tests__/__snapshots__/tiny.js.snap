// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass glam object prop 1`] = `
.glamor-0,
[data-glamor-0] {
  margin-left: 24px;
  font-size: 20px;
}

<div
  class="glamor-0"
  id="hey-there"
/>
`;
